packages/core/EventManager.ts

```diff
@@ -377,36 +379,49 @@ export default class EventManager {
     event: CalendarEvent,
     booking: PartialBooking
   ): Promise<Array<EventResult<NewCalendarEventType>>> {
-    return async.mapLimit(this.calendarCredentials, 5, async (credential: Credential) => {
-      try {
-        // HACK:
-        // Right now if two calendars are connected and a booking is created it has two bookingReferences, one is having uid null and the other is having valid uid.
-        // I don't know why yet - But we should work on fixing that. But even after the fix as there can be multiple references in an existing booking the following ref.uid check would still be required
-        // We should ignore the one with uid null, the other one is valid.
-        // Also, we should store(if not already) that which is the calendarCredential for the valid bookingReference, instead of going through all credentials one by one
-        const [bookingRef] = booking.references
-          ? booking.references.filter((ref) => ref.type === credential.type && !!ref.uid)
-          : [];
-
-        if (!bookingRef) throw new Error("bookingRef");
-
-        const { uid: bookingRefUid, externalCalendarId: bookingExternalCalendarId } = bookingRef;
-
-        if (!bookingExternalCalendarId) throw new Error("externalCalendarId");
-
-        return updateEvent(credential, event, bookingRefUid, bookingExternalCalendarId);
-      } catch (error) {
-        let message = `Tried to 'updateAllCalendarEvents' but there was no '{thing}' for '${credential.type}', userId: '${credential.userId}', bookingId: '${booking.id}'`;
-        if (error instanceof Error) message = message.replace("{thing}", error.message);
-        console.error(message);
-        return Promise.resolve({
-          type: credential.type,
+    try {
+      // Bookings should only have one calendar reference
+      const calendarReference = booking.references.filter((reference) =>
+        reference.type.includes("_calendar")
+      )[0];
```

Comment:
Bookings are only created on one calendar so we just grab the first result from filter